### ISSUE TITLE
ARTS2D: make 2-D susceptibility consistent with 1-D and correct drift geometry

### ISSUE THREAD

**Issue by joglekara** [ergodicio/tsadar#135]
## Problem

The ARTS 2-D electron susceptibility is inconsistent with the 1-D implementation in two independent ways.

### 1. Imaginary sign

The 1-D path uses

```python
chiEI = -jnp.pi / (klde**2) * 1j * df
```

at `tsadar/core/physics/form_factor.py:316`, while the 2-D helper returns `+pi / klde^2 * dfe` at line 631 and inserts it as `+1j * chiEI` at line 820.

For an isotropic Maxwellian, matched projections at normalized phase velocities 0.5, 1, and 2 give matching projected EDF and real principal-value susceptibility, but `Im(chi_2D) / Im(chi_1D)` is approximately `-1` at all three points. The 1-D sign was intentionally corrected in Avi-authored commit `242e4ed0`; the 2-D path was missed.

This sign bug is real but is not by itself the small-angle resonance fix: conjugating an electron-only `epsilon` leaves `|epsilon|^2` unchanged.

### 2. Drift-dependent projection geometry

At `form_factor.py:809-818`, the 2-D path constructs `(omega * k / k^2 - ud) / vTe`, replaces the signed resonance coordinate by its magnitude, and derives `beta` from that drift-shifted vector using `atan(y/x)`.

For longitudinal susceptibility, the projection direction should remain `khat`; drift enters the signed scalar resonance

```text
xi = (omega / |k| - ue dot khat) / vTe
```

A drift component perpendicular to `k` must not rotate the Radon projection. `atan(y/x)` also has an undefined `0/0` origin and NaN gradients on an axis; use `atan2` where an angle is required and define the zero-vector limit.

The current `project` docstring says `beta` is the k-vector angle, which conflicts with its caller. The `Va` and `ud` frame semantics also differ between the 1-D and 2-D paths.

## Acceptance criteria

- [ ] Benchmark 1-D and 2-D electron susceptibility against an analytic Maxwellian/Faddeeva reference over positive and negative phase velocities; real and imaginary parts use one documented Fourier/Landau convention and agree within discretization error.
- [ ] For an isotropic zero-flow Maxwellian, 1-D and 2-D spectra agree within a documented tolerance.
- [ ] Compute the projection direction from `khat` with `atan2` and retain a signed `xi`.
- [ ] Purely perpendicular drift leaves the projected susceptibility unchanged; parallel drift gives only the expected Doppler translation.
- [ ] Add an analytic shifted anisotropic-Gaussian Radon test covering mean, variance, coordinate signs, axis-aligned `k`, `xi=0`, and finite gradients.
- [ ] Document and test the `Va`/`ud` frame convention, including multi-ion cases, so 1-D and 2-D use the same Galilean transformation.

Related: #124.

**Comment by almilder** [comment]
Traced through problem 2 in `calc_in_2D` / `calc_chi_vals` — agree it's a real bug, but want to refine the framing a bit before we scope the fix.

**Beta rotating with `ud` is a genuine bug, and it's isolated to one line.** The scalar Doppler shift itself is fine — `omgdop = omg - vdot(k, Va)` (`form_factor.py:791`) correctly dots the *ion* flow vector into `k`. The problem is in how the *electron* resonance vector is built at `form_factor.py:814`:

```python
xie = vdiv(vsub(vdot(omgdop / k_mag**2, k), ud), vTe)
```

`vdot(omgdop / k_mag**2, k)` = `(omgdop/|k|) * k̂`, which is correctly a vector along `k̂`. But the *full* 2D `ud` is then subtracted from it, not just `ud`'s projection onto `k̂`. Any `u_perp` component rotates the result away from `k̂`, and since `beta` is derived from this rotated vector's direction, the projection axis moves with drift when it should stay locked to `k̂` regardless of drift.

**The "unsigned `xie_mag`" framing undersells what's actually wrong — no sign is lost in the magnitude/angle split itself.** Downstream in `calc_chi_vals` (`form_factor.py:606-633`), `fe_vphi`, `dfe`, and the `ratintn` pole location are all evaluated at `xie_mag_at` along the profile projected at angle `beta`. Since `(xie_mag, beta)` is just the polar form of `xie`, `proj_beta(xie_mag)` reconstructs the correct value at the point `xie` — `beta` carries the orientation that a sign would otherwise encode, so nothing is discarded by taking the magnitude. In other words, "unsigned `xie_mag`" is a symptom of point 1, not an independent defect: once `xie` is built correctly (k-only direction, drift entering only as a scalar shift), the magnitude/angle decomposition of it is already fine as-is. Suggest we drop the "signed scalar" framing from the acceptance criteria and state the actual requirement instead: `beta` must be derived from `k̂` alone, and the resonance coordinate must use only the k-projection of drift, `ud·k̂`.

**`atan2` is the right fix for the quadrant issue, with one caveat.** `jnp.arctan(xie[1]/xie[0])` plus the `heaviside` term at `form_factor.py:830` is a hand-rolled 4-quadrant `atan2`, so replacing it directly with `atan2(y, x)` removes both the `0/0` division and the NaN-gradient blowup on the `x=0` axis. `atan2(0, 0)` is still mathematically undefined at the true origin though, so we still need to explicitly pick and document a convention for the zero-vector case (e.g. `beta = 0`, since the projected value there should be direction-independent by symmetry — worth confirming that before locking it in).

**Proposed fix:**

```python
k_hat = k / k_mag                       # unchanged by drift
beta = atan2(k_hat[1], k_hat[0])        # projection direction: k alone
ud_par = vdot(ud, k_hat)                # only the k-projection of drift
xi = (omgdop / k_mag - ud_par) / vTe    # signed scalar, evaluation point
```

with the sign of `xi` folded into which side of the `beta`-projected profile gets sampled (either via `beta + pi` when `xi < 0`, or by having `_interp_beta_v` accept a signed coordinate directly rather than routing through a separate magnitude/angle split). This directly satisfies the "perpendicular drift leaves the projection unchanged" and "parallel drift gives only the expected Doppler translation" acceptance criteria, since `beta` no longer depends on drift at all.
